src/main/java/edu/cornell/kfs/fp/batch/xml/cloudcheckr/CloudCheckrWrapper.java

```diff
@@ -0,0 +1,135 @@
+package edu.cornell.kfs.fp.batch.xml.cloudcheckr;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlElementWrapper;
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import edu.cornell.kfs.sys.xmladapters.KualiDecimalXmlAdapter;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "GetDetailedBillingWithGroupingResponseV2", namespace = StringUtils.EMPTY)
+public class CloudCheckrWrapper {
+    @XmlElement(name = "Total", namespace = StringUtils.EMPTY)
+    @XmlJavaTypeAdapter(KualiDecimalXmlAdapter.class)
+    private KualiDecimal total;
+    
+    @XmlElement(name = "Max", namespace = StringUtils.EMPTY)
+    @XmlJavaTypeAdapter(KualiDecimalXmlAdapter.class)
+    private KualiDecimal maximum;
+    
+    @XmlElement(name = "Min", namespace = StringUtils.EMPTY)
+    @XmlJavaTypeAdapter(KualiDecimalXmlAdapter.class)
+    private KualiDecimal minimum;
+    
+    @XmlElement(name = "Average", namespace = StringUtils.EMPTY)
+    @XmlJavaTypeAdapter(KualiDecimalXmlAdapter.class)
+    private KualiDecimal average;
+    
+    @XmlElementWrapper(name = "CostsByGroup", namespace = StringUtils.EMPTY)
+    @XmlElement(name = "GroupingLevel", namespace = StringUtils.EMPTY, required = true)
+    private List<GroupLevel> groupLevels;
```

Comment:
I am wondering if 'costsByTime' may be a better name for this list?

Reply:
How about costsByGroup? .... sure I can change this

Reply:
that is what I meant but must have copy pasted the one below this. thank you.